### PR TITLE
Feature BLO-68 and BLO-70  UI UX + Redux implementation

### PR DESCRIPTION
**Pull request by roelvdboom**
Also contains some work from BLO-70.

Login
image

Profile
image

Create Transaction (Missing create button)
image

Transaction detail
image

### CODE REVIEW

File: src/__tests__/components/login.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        configure({ adapter: new Adapter() });
+        const store = mockStore(initialState);
+        const container = shallow(
+            <Provider store={store}>
+                <Login />
+            </Provider>,
+        ).dive();
+        expect(Object.keys(container.props()).length).toEqual(1);
+        // TODO: Test if state is set correctly
```
Comment (timofudala) [inline src/__tests__/components/login.test.tsx]: Please implement this TODO.

File: src/components/login/login.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        });
+    };
+
+    public handlePrivateKeyChange = (element) => {
+        this.setState({
+            privateKey: element.target.value,
+        });
+    };
+
+    public login = () => {
```
Comment (timofudala) [inline src/components/login/login.tsx]: Please conform to the React naming of `handle` in front of the actual name, so: `handleLogin`.
Please apply this throughout the entire application.

File: src/services/apiService.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { logger } from "@blockr/blockr-logger";
+import { Transaction } from "@blockr/blockr-models";
+import Axios from "axios";
+
+const publicApiUrl = "https://public.blockr.verux.nl";
```
Comment (timofudala) [inline src/services/apiService.ts]: Please make these constants outside the scope of a class with upper case, so `PUBLIC_API_URL`.

File: src/services/apiService.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const transactionRoute = publicApiUrl + "/transactions";
+
+export class ApiService {
+    public getAllTransactionsAsync = (): Promise<Transaction[]> => {
+        logger.info("Within api Service");
```
Comment (timofudala) [inline src/services/apiService.ts]: Please remove this logging.

**Review by timofudala — CHANGES_REQUESTED** [review]

File: src/components/profile/profile.tsx [hunk 5]
```diff
@@ -49,12 +48,12 @@ class Profile extends React.Component<Props, {}> {
...
     }
 
-    public logout = () => {
+    public handleLogout = () => {
         this.props.logout();
         goToUrl("/");
     };
 
-    public goToTransactionDetailView = (transaction: Transaction) => {
+    public handleTransactionView = (transaction: Transaction) => {
```
Comment (timofudala) [inline src/components/profile/profile.tsx:52]: I appreciate you conforming to the `handle` aspect now. However, please change this function name to `handleTransactionViewNavigation`.

**Review by timofudala — DISMISSED** [review]

File: src/components/createTransaction/createTransaction.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import { Transaction, TransactionType } from "@blockr/blockr-models";
```
Comment (timofudala) [inline src/components/createTransaction/createTransaction.tsx]: Your file naming is inconsistent within this folder and potentially in other folders as well.

**Review by saltz — DISMISSED** [review]
Remove the jest section from the `package.json` file. And if published add a `.npmignore` file.

File: Jenkinsfile [hunk 7]
```diff
@@ -7,7 +7,10 @@ String repo = "blockr-wallet"
 Map settings = [
     sonar_key: "blockr-wallet",
     source_folder: "src/",
-    archive_folders: ["dist/", "release/"]
+    archive_folders: ["dist/", "release/"],
+    yarn: true,
+    skip_tests: true,
+    sonar_exclusions: "src/*"
```
Comment (Juuldamen) [inline Jenkinsfile]: Why are all source files excluded?

File: src/components/application.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import "react-toastify/dist/ReactToastify.css";
+import { Grid, Segment } from "semantic-ui-react";
+import "../global.scss";
+import routerHistory from "../store/routerHistory";
+import CreateTransaction from "./createTransaction/createTransaction";
+import Login from "./login/login";
+import Profile from "./profile/profile";
+import Transaction from "./transaction/transaction";
+
+// let validatorIp: string | undefined;
```
Comment (hugovandevliert) [inline src/components/application.tsx]: Please remove this commented out code

**Review by RikvanSpreuwel — APPROVED** [review]

File: src/main.ts [hunk 9]
```diff
@@ -1,56 +1,56 @@
...
 
 let win: BrowserWindow | null;
 
 const installExtensions = async () => {
-    const installer = require('electron-devtools-installer');
     const forceDownload = !!process.env.UPGRADE_EXTENSIONS;
-    const extensions = ['REACT_DEVELOPER_TOOLS', 'REDUX_DEVTOOLS'];
+    const extensions = ["REACT_DEVELOPER_TOOLS", "REDUX_DEVTOOLS"];
 
     // TODO: Add catch
```
Comment (hugovandevliert) [inline src/main.ts]: Please finish this TODO

**Review by hugovandevliert — DISMISSED** [review]

File: src/main.ts (hunk 9 above)
Comment (Juuldamen) [inline src/main.ts]: Resolve this TODO.

File: src/reducers/index.ts [hunk 10]
```diff
@@ -1,11 +1,14 @@
-import { Reducer, combineReducers } from 'redux';
+import { combineReducers, Reducer } from "redux";
```
Comment (Juuldamen) [inline src/reducers/index.ts]: Remove this unused import.

```suggestion
import { combineReducers } from "redux";
```

**Review by Juuldamen — CHANGES_REQUESTED** [review]

**Review by Juuldamen — APPROVED** [review]

**Pull request merged**